### PR TITLE
[Android] Dependency Update and Cleanup

### PR DESCRIPTION
**Pull request by Tsunamical**
If there are any issues as always, let me know.

[Review on Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/3702)

### CODE REVIEW

**Comment by Tsunamical** [comment]
Don't know if you'd like the last commit to be split up any or not. It does combine a lot of things - switches to foreach loops, inlining of variables, removing redundant variable setting, reducing ArrayList verbosity, fixing an unclosed cursor, etc. If there are any questions or concerns, let me know. I somehow managed to accidentally remove the "This change is reviewable" button lol

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlay.java [hunk 1]
```diff
@@ -277,14 +276,11 @@ private static InputOverlayDrawableJoystick initializeOverlayJoystick(Context co
...
-		final InputOverlayDrawableJoystick overlayDrawable
-				= new InputOverlayDrawableJoystick(res,
-					bitmapOuter, bitmapInner,
-					outerRect, innerRect,
-					joystick);
 
-
-		return overlayDrawable;
+		return new InputOverlayDrawableJoystick(res,
+            bitmapOuter, bitmapInner,
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlay.java]: Indentation mismatch here. We use tabs for indenting and spaces for vertical alignment

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/EGLHelper.java [hunk 2]
```diff
@@ -120,14 +120,13 @@ public void closeHelper()
 	 */
 	public String[] getEGLInfo()
 	{
-		String[] info = {
+
+		return new String[]{
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/EGLHelper.java]: Space after the `]`

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/Java_GCAdapter.java [hunk 3]
```diff
@@ -77,41 +77,34 @@ public static void InitAdapter()
...
 
 	public static boolean OpenAdapter()
 	{
 		HashMap<String, UsbDevice> devices = manager.getDeviceList();
-		Iterator it = devices.entrySet().iterator();
-		while (it.hasNext())
-		{
-			HashMap.Entry pair = (HashMap.Entry)it.next();
-			UsbDevice dev = (UsbDevice)pair.getValue();
+		for (Object o : devices.entrySet()) {
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/Java_GCAdapter.java]: `for (Map.Entry<String, UsbDevice> entry : device.entrySet())`

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/Java_GCAdapter.java [hunk 4]
```diff
@@ -77,41 +77,34 @@ public static void InitAdapter()
...
+			HashMap.Entry pair = (HashMap.Entry) o;
+			UsbDevice dev = (UsbDevice) pair.getValue();
 			if (dev.getProductId() == 0x0337 && dev.getVendorId() == 0x057e) {
-				if (manager.hasPermission(dev))
-				{
+				if (manager.hasPermission(dev)) {
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/Java_GCAdapter.java]: Our coding style uses braces on the next line

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlay.java (hunk 1 above)
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlay.java]: Indentation mismatch. We use tabs for indentation and spaces for vertical alignment

**Comment by lioncash** [comment]
Note: Just because your IDE makes an inspection correction suggestion, it doesn't always mean it's a good idea.

**Comment by Tsunamical** [comment]
Right, I was choosing things that seemed to be relevant and beneficial (some things were not suggested by the IDE either though). The one I'm most conflicted on is the LIBUSB change though, I can't properly test as my current device in my hands is an ARMv7 device, hence the asking for criticism. Fixing now, thanks.

**Comment by Tsunamical** [comment]
Review status: 0 of 24 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/EGLHelper.java, line 124 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3702#-KBv9nI47pzF5nABkeCb-r1-124:-KBzIjkZSNOHI3bqKLXR:-499146009) ([raw file](https://github.com/dolphin-emu/dolphin/blob/8dd588722b9091a1ee1fee4923a3b7d8b14b8cf3/Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/EGLHelper.java#L124)):*
Done.

---

*[Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/Java_GCAdapter.java, line 90 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3702#-KBv9nI47pzF5nABkeCc-r1-90:-KBzIl4kZUJfCLwGT7sp:-499146009) ([raw file](https://github.com/dolphin-emu/dolphin/blob/8dd588722b9091a1ee1fee4923a3b7d8b14b8cf3/Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/Java_GCAdapter.java#L90)):*
Done.

---

*[Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/Java_GCAdapter.java, line 94 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3702#-KBv9nI47pzF5nABkeCc-r1-94:-KBzIlhispuqhqp6Fcod:-499146009) ([raw file](https://github.com/dolphin-emu/dolphin/blob/8dd588722b9091a1ee1fee4923a3b7d8b14b8cf3/Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/Java_GCAdapter.java#L94)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3702)*

**Comment by Tsunamical** [comment]
Added a few more commits.

**Comment by lioncash** [comment]
Pinging @sigmabeta to also review this

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/Game.java [hunk 5]
```diff
@@ -26,7 +26,7 @@
 	public static final int COUNTRY_WORLD = 12;
 	public static final int COUNTRY_UNKNOWN = 13;
 
-	private static final String PATH_SCREENSHOT_FOLDER = "file:///sdcard/dolphin-emu/ScreenShots/";
+	private static final String PATH_SCREENSHOT_FOLDER = android.os.Environment.getExternalStorageDirectory().getPath() + "/dolphin-emu/ScreenShots/";
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/Game.java]: Please use an import to shorten this

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/NativeLibrary.java [hunk 6]
```diff
@@ -352,7 +352,7 @@ private NativeLibrary()
 
 	/**
 	 * Enables or disables CPU block profiling
-	 * @param enable
+	 * @param enable  Enable or disable profiling.
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/NativeLibrary.java]: one space too many here.

Comment (Tsunamical) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/NativeLibrary.java]: nnn Was looking at ones above it, my bad

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/Game.java (hunk 5 above)
Comment (Tsunamical) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/Game.java]: Done

**Comment by Tsunamical** [comment]
I believe that the loop in OpenAdapter can be simplified down to what we use generically in the rest of the class - but someone correct me if I'm wrong.

**Comment by sigmabeta** [comment]
@dolphin-emu-bot rebuild

**Comment by Tsunamical** [comment]
Huh, looks like on bintray they don't have 23.2.0 yet? Guess we'll roll back to 23.1.1 then.

**Comment by sigmabeta** [comment]
The Android Support Libraries are not ever, to my knowledge, hosted on Bintray.

In order for this to build correctly, we have to get @Parlane to run the Android SDK dependency update script. 

Once that's done though, this is fine to merge IMO. A lot of common sense fixes. Thanks for doing this.

**Comment by Tsunamical** [comment]
Oops, you're right, I'm silly. Thanks!

**Comment by sigmabeta** [comment]
Quick pro-tip: if you are using IntelliJ, `Source/Android/code-style-java.jar` contains a code-style template you can import. Then have it run the auto-formatter when you make your commits.

I forget to do this almost literally every time though, so I won't get too mad if you do also.

**Comment by Tsunamical** [comment]
Ah, that's pretty handy. I'm using Android Studio, so I'll going to assume it's available on there too.

**Comment by sigmabeta** [comment]
It is; when I say IntelliJ really I mean Android Studio, but I've known some Android devs who really do use IntelliJ, so I like to use the more general name for the sake of inclusiveness.

**Comment by Tsunamical** [comment]
Right, thanks! That's pretty neat, thanks again. Can you ping me once the SDK is updated?

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/Java_WiimoteAdapter.java [hunk 7]
```diff
@@ -86,13 +86,12 @@ public static int Output(int index, byte[] buf, int size)
 
 		final int LIBUSB_REQUEST_TYPE_CLASS  = (1 << 5);
 		final int LIBUSB_RECIPIENT_INTERFACE = 0x1;
-		final int LIBUSB_ENDPOINT_OUT        = 0;
```
Comment (Tsunamical) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/Java_WiimoteAdapter.java]: Even though this is a pointless bitwise calculation, do we want to leave it as an indicator?

**Comment by Tsunamical** [comment]
Going to rebase and resubmit once more due to Git screwing up my branch completely, and will remove some resounding issues still. I assume after the previous update we will be sticking to 1.5.0 for the Gradle plugin? I believe 1.5.1 is the latest version available in the stable channel.

**Pull request closed without merging**